addons/mail/static/tests/chatter_tests.js

```diff
@@ -371,6 +371,77 @@ QUnit.test('list activity exception widget with activity', async function (asser
     list.destroy();
 });
 
+QUnit.test("list activity widget : displays notification to save record before scheduling activity", async function (assert) {
+    assert.expect(9);
+
+    const currentUser = this.data["res.users"].records.find(
+        (user) => user.id === this.data.currentUserId
+    );
+    Object.assign(currentUser, {
+        activity_ids: [1, 4],
+        activity_state: "today",
+        activity_type_id: 3,
+        activity_type_icon: "fa-phone",
+    });
+
+    const { widget: list } = await start({
+        hasView: true,
+        View: ListView,
+        model: "res.users",
+        data: this.data,
+        arch:
+            "<list editable='bottom'>"+
+            "<field name='foo'/>"+
+            "<field name='activity_ids' widget='list_activity'/>"+
+            "</list>",
+        mockRPC: function (route) {
+            if (!["/mail/init_messaging", "/mail/load_message_failures"].includes(route)) {
+                assert.step(route);
+            }
+            return this._super(...arguments);
+        },
+        services: {
+            notification: {
+                notify(notification) {
+                    assert.strictEqual(
+                        notification.message,
+                        "Please save the record before scheduling an activity.",
+                        "should display the alert information to save the record before scheduling activity"
+                    );
+                },
+            },
+        },
+    });
+
+    const $firstRow = list.$(".o_data_row:first");
```

Comment:
Test is far too long, covers too many things, and it's gonna be quite annoying to forwardport.
Also the selectors are poor, e.g. `o_activity_color_default` for selecting the dirty activity widget being selected, `fa-phone` being totally useless for the actual fix, etc.

It's much better to keep test very small and to the point, otherwise the test is hardly maintainable.